refactor combine.ts

Soolan opened the pull request:
- replace lodash methods with native js
- simplify the code
- add a new method for transaction validation

intelliot commented:
This will fix https://github.com/ripple/ripple-lib/issues/987

FKSRipple commented on src/transaction/combine.ts:
```diff
@@ -1,39 +1,27 @@
+    validate.combine({signedTransactions})
+    const transactions: any[] = signedTransactions.map(binary.decode)
+    const tx = transactions[0]
+    delete tx.Signers
```
you can remove this line, since you delete tx.Signers a few lines later

FKSRipple commented on src/transaction/combine.ts:
```diff
@@ -1,39 +1,27 @@
+    const tx = transactions[0]
+    delete tx.Signers
+    transactions.forEach( _tx => {
+        delete _tx.Signers;
```
can you add a comment here for why you're doing this?

FKSRipple commented on src/transaction/combine.ts:
```diff
@@ -1,39 +1,27 @@
-  const signedTransaction = binary.encode(signedTx)
-  const id = computeBinaryTransactionHash(signedTransaction)
-  return {signedTransaction, id}
+    const txs = hashArrayToObjectArray(signedTransactions)
```
nit: seems like the indentation is off

Soolan commented:
> @Soolan thanks for the PR, I left a few comments. I know it's been a while, but are you still interested in getting this merged? Let me know and I'll help get it merged

Hi @FKSRipple 
Thank you for your response.
By all means please help me to merge that.


p.s.
I did study/review the code a while ago, and contributed tiny bits here and there, hoping to get the job Ripple advertised a while back. 
I would LOVE to work on this repository on a more serious level. (contract, permanent, anything! as long as I am contributing to Ripple I am happy).

If your offer still stands, please give me 8 days to deliver my current contract and then lets talk about it.

Thank you.

In the mean time, I wonder if you are kind enough to take care of the changes please. I agree with all your comments.

FKSRipple commented:
@Soolan if you grant me access to edit the files on your branch, I'll make the recommended changes.

/cc @intelliot re: contracting

Soolan commented:
So sorry @FKSRipple 
Last month I did a big cleanup on all my repos and deleted my ripple-lib fork as well.
I can't restore it because github keeps only (non-forked) deleted repos (for 90 days).

I've deleted my local files as well.
Give me a few days to finish my current task and If this issue was still open by then I will do it again and will apply your reviews.
Sorry again and thank you.

FKSRipple commented:
@Soolan np, I'll copy the instructions that I see in GitHub to check out the branch, it seems like it's still stored somewhere:

```
git checkout -b Soolan-987-cleanup-combine develop
curl -L https://github.com/ripple/ripple-lib/pull/1014.patch | git am -3
```

FKSRipple commented:
Closing this PR but feel free to resubmit using the `git checkout` instructions above to get back your old branch
